pkg/prometheus/promcfg.go

```diff
@@ -535,6 +535,18 @@ func mergeAttachMetadataWithScrapeClass(attachMetadata *monitoringv1.AttachMetad
 	}
 }
 
+func mergeFallbackScrapeProtocolWithScrapeClass(fallbackScrapeProtocol *monitoringv1.ScrapeProtocol, scrapeClass monitoringv1.ScrapeClass) *monitoringv1.ScrapeProtocol {
+	if fallbackScrapeProtocol == nil {
+		fallbackScrapeProtocol = scrapeClass.FallbackScrapeProtocol
+	}
+
+	if fallbackScrapeProtocol == nil {
+		return nil
+	}
```

Comment:
```suggestion
```
Redundant code, can it be removed?

Reply:
not redundant code, first statement checks fallbackScrapeProtocol is set or not. Since fallbackScrapeProtocol is set from ScrapeClass in that case.

So in second statetment fallbackScrapeProtocol == nil actually would be checking if it was nil even in scrapeClass

Reply:
I think that we can remove the test though since the next line will return  fallbackScrapeProtocol.